### PR TITLE
Manage Panel HPV Tests

### PR DESCRIPTION
**Pull request by mickohanlon23**
Fixes the following issue: 

> Our CDS assumes that if a patient is HPV16+, they will have a general LOINC code from the [HPV Test](https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/blob/37-add-obs-bug-fix/cql/DashboardLibrary.cql#L78) value set, and then it will specifically have a valueCodeableConcept result SNOMED code of [HPV16+](https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/blob/37-add-obs-bug-fix/cql/DashboardLibrary.cql#L110).
In reality, with the way UMMC's panel tests are formatted, if a patient is HPV16+, they will have a LOINC code that is specific to HPV16, and then a result SNOMED code indicating "Positive".

This issue was realized as a result of updates that were made in PR https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/116. This update modifies the CollateConclusionCodes function so that if the function is being called to collate an HPV test that has it's result in an Observation, if that Observation has a `code` to represent HPV16 or HPV18, and a `valueCodeableConcept` that just represents a generic positive result (ex. "Detected"), then the function will return the value of HPV16+ or HPV18+ as opposed to just the generic positive value. Previously, our CDS was seeing the generic positive value, and counting the value as generic "HPV+", even though that positive result was for a more severe test of HPV16 or HPV18.

### CODE REVIEW

File: cql/DashboardLibrary.cql [hunk 1]
```diff
@@ -786,13 +788,47 @@ define function CollateConclusionCodes(D DiagnosticReport, ObsList List<Observat
   else
     ValuesFromObservation(D, ObsList)
 
+define function ToFhirCodeableConcept(c System.Code):
```
Comment (yunwwang) [inline cql/DashboardLibrary.cql:791]: I think this function worth to put in FHIRHelpers. This is helpful if we need do similar convert in other modules.

File: cql/DashboardLibrary.cql [hunk 2]
```diff
@@ -786,13 +788,47 @@ define function CollateConclusionCodes(D DiagnosticReport, ObsList List<Observat
...
+// @returns {FHIR.CodeableConcept} the valueCodeableConcept of the Observation,
+// or the CodeableConcept of HPV16+ or HPV18+ if the test was positive for code of HPV16 or HPV18
+define function ConvertHpvTestTypeToValue(O Observation):
+  if O.value as FHIR.CodeableConcept in "High Risk HPV Positive Results" then
+    if O.code in "HPV 16 Tests" then 
+      Hpv16CodeToFhirCodeableConcept
+    else if O.code in "HPV 18 Tests" then 
+      Hpv18CodeToFHIRCodeableConcept
+    else
+      O.value
```
Comment (yunwwang) [inline cql/DashboardLibrary.cql:821]: convert to CodeableConcept to match the return type.

File: cql/DashboardLibrary.cql [hunk 3]
```diff
@@ -786,13 +788,47 @@ define function CollateConclusionCodes(D DiagnosticReport, ObsList List<Observat
...
+define function ConvertHpvTestTypeToValue(O Observation):
+  if O.value as FHIR.CodeableConcept in "High Risk HPV Positive Results" then
+    if O.code in "HPV 16 Tests" then 
+      Hpv16CodeToFhirCodeableConcept
+    else if O.code in "HPV 18 Tests" then 
+      Hpv18CodeToFHIRCodeableConcept
+    else
+      O.value 
+  else 
+    O.value
```
Comment (yunwwang) [inline cql/DashboardLibrary.cql:823]: convert to CodeableConcept to match the return type.

File: cql/DashboardLibrary.cql [hunk 4]
```diff
@@ -786,13 +788,47 @@ define function CollateConclusionCodes(D DiagnosticReport, ObsList List<Observat
...
+define Hpv18CodeToFHIRCodeableConcept:
+  ToFhirCodeableConcept("HPV18+")
+
+// If an HPV test has a code for a genotyped high-risk HPV test, and a generic value representing a positive test,
+// this function returns the valueCodeableConcept that would represent the positive genotyped high-risk test,
+// as opposed to the generic positive value
+// @param O - a FHIR Observation
+// @returns {FHIR.CodeableConcept} the valueCodeableConcept of the Observation,
+// or the CodeableConcept of HPV16+ or HPV18+ if the test was positive for code of HPV16 or HPV18
+define function ConvertHpvTestTypeToValue(O Observation):
```
Comment (yunwwang) [inline cql/DashboardLibrary.cql:814]: Change function name to ConvertHpvTestTypeToCodeableConcept to match the return type.

File: cql/DashboardLibrary.cql (hunk 1 above)
Comment (mickohanlon23) [reply cql/DashboardLibrary.cql]: Good idea, I think it is generic enough to move there. I did this in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/117/commits/938d71d318b5a23471dc3c0d8a7c1b81b3eebe2e.

File: cql/DashboardLibrary.cql (hunk 3 above)
Comment (mickohanlon23) [reply cql/DashboardLibrary.cql]: Done: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/117/commits/938d71d318b5a23471dc3c0d8a7c1b81b3eebe2e. I wonder if that could have been casing the issue we were seeing with UMMC.

File: cql/DashboardLibrary.cql (hunk 2 above)
Comment (mickohanlon23) [reply cql/DashboardLibrary.cql]: Also done: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/117/commits/938d71d318b5a23471dc3c0d8a7c1b81b3eebe2e.

File: cql/DashboardLibrary.cql (hunk 4 above)
Comment (mickohanlon23) [reply cql/DashboardLibrary.cql]: Makes sense. I did this in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/117/commits/938d71d318b5a23471dc3c0d8a7c1b81b3eebe2e.

**Review by yunwwang — APPROVED** [review]

**Pull request merged**